### PR TITLE
daemon: reach for Overlord.Loop less thanks to overlord.Mock

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

**Comment by pedronis** [comment]
based on #3890

File: daemon/api_test.go [hunk 1]
```diff
@@ -292,6 +294,71 @@ func (s *apiBaseSuite) daemon(c *check.C) *Daemon {
...
+
+func (s *apiBaseSuite) daemonWithFakeSnapManager(c *check.C) *Daemon {
+	d := s.daemonWithOverlordMock(c)
+	st := d.overlord.State()
+	d.overlord.AddManager(newFakeSnapManager(st))
+	return d
+}
+
+func (s *apiBaseSuite) waitTrivialChange(c *check.C, chg *state.Change) {
+	err := s.d.overlord.Settle(2 * time.Second)
```
Comment (mvo5) [inline daemon/api_test.go]: Should this be settleTimeout (15s) like in the other tests?

Comment (pedronis) [reply daemon/api_test.go]: not especially, this tests deal with just one task,  the tests we had to bump to 15 under overlord are all complex tasks like device reg or first boot or multiple snap refreshes

Comment (pedronis) [reply daemon/api_test.go]: done

Comment (pedronis) [reply daemon/api_test.go]: to clear I'm happy to bump these numbers again  if we find that they are still too low, but if 5s is not enough here, then probably 15s is not enough for those other more complicated tests

File: daemon/api_test.go [hunk 2]
```diff
@@ -292,6 +294,74 @@ func (s *apiBaseSuite) daemon(c *check.C) *Daemon {
...
+}
+
+type fakeSnapManager struct {
+	runner *state.TaskRunner
+}
+
+func newFakeSnapManager(st *state.State) *fakeSnapManager {
+	runner := state.NewTaskRunner(st)
+
+	runner.AddHandler("fake-install-snap", func(t *state.Task, _ *tomb.Tomb) error {
```
Comment (mvo5) [inline daemon/api_test.go]: Nice that this is isolated in tests now!

**Pull request merged**